framework/Optimizers/stepManipulators/GradientHistory.py

```diff
@@ -139,11 +158,18 @@ def step(self, prevOpt, gradientHist=None, prevStepSize=None, recommend=None, **
     """
     stepSize = self._stepSize(gradientHist=gradientHist, prevStepSize=prevStepSize,
                               recommend=recommend, **kwargs)
-    gradient = gradientHist[-1][1]
+    direction = dict((var, 0) for var in self._optVars)
+    for t in range(self._gradTerms):
+      if len(gradientHist) <= t:
+        break
+      decay = np.exp(-t * self._termDecay)
+      for var in direction:
+        direction[var] -= gradientHist[-(t+1)][1][var] * decay
+    # gradient = gradientHist[-1][1]
     # use gradient, prev point, and step size to choose new point
     newOpt = {}
     for var in self._optVars:
-      newOpt[var] = prevOpt[var] - stepSize * gradient[var]
+      newOpt[var] = prevOpt[var] + stepSize * direction[var]
```

Comment:
I like that you changed this to direction instead of gradient because it's more accurate. step size times the direction (gradient unit vector not gradient).